### PR TITLE
Remove use of isTransit() function

### PR DESCRIPTION
**Pull request by daniel-heppner-ibigroup**
**Description:**
The isTransit() function is unneeded because we can just use the `transitLeg` propery on legs. It was causing bugs with some of our custom modes.

### CODE REVIEW

**Comment by daniel-heppner-ibigroup** [comment]
I also deleted a file that wasn't imported anywhere. Did we mean to be using that?

**Comment by miles-grant-ibigroup** [comment]
> I also deleted a file that wasn't imported anywhere. Did we mean to be using that?

That's from the tabbed UI. Nobody that I know of currently uses that but technically it's still supported? What do we think

**Review by miles-grant-ibigroup — APPROVED** [review]
We're really on OTP2 now... eek!

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Updating my review here to continue using the `isTransit` method in core-utils instead. Let's refactor that! This PR shouldn't be closed though I like the cleanup

**Comment by daniel-heppner-ibigroup** [comment]
> > I also deleted a file that wasn't imported anywhere. Did we mean to be using that?
> 
> That's from the tabbed UI. Nobody that I know of currently uses that but technically it's still supported? What do we think

Well it wasn't even exported in index.js

**Comment by miles-grant-ibigroup** [comment]
> > > I also deleted a file that wasn't imported anywhere. Did we mean to be using that?
> > 
> > 
> > That's from the tabbed UI. Nobody that I know of currently uses that but technically it's still supported? What do we think
> 
> Well it wasn't even exported in index.js

Let's just update core-utils instead

**Comment by daniel-heppner-ibigroup** [comment]
Will block this one until the OTP UI side is merged

EDIT: ready to go!

**Review by miles-grant-ibigroup — APPROVED** [review]
Swaggy

File: lib/actions/field-trip.js
Comment (miles-grant-ibigroup) [inline lib/actions/field-trip.js]:
```suggestion
        .filter((leg) => isTransitLeg(leg))
```

File: lib/components/narrative/default/default-itinerary.js
Comment (miles-grant-ibigroup) [inline lib/components/narrative/default/default-itinerary.js]:
```suggestion
      if (isTransitLeg(leg)) {
```

File: lib/components/narrative/default/itinerary-description.tsx
Comment (miles-grant-ibigroup) [inline lib/components/narrative/default/itinerary-description.tsx:35]: I'm scared to remove the current check

File: lib/components/narrative/default/itinerary-summary.js
Comment (miles-grant-ibigroup) [inline lib/components/narrative/default/itinerary-summary.js:100]: I'd like to keep the old check maybe

File: lib/components/narrative/line-itin/LegIconWithA11y.tsx
Comment (miles-grant-ibigroup) [inline lib/components/narrative/line-itin/LegIconWithA11y.tsx]: Why aren't we using the new method

File: lib/components/narrative/line-itin/realtime-time-column.tsx
Comment (miles-grant-ibigroup) [inline lib/components/narrative/line-itin/realtime-time-column.tsx]: This is now checking for the existence of a method

File: lib/components/narrative/narrative-itineraries.js
Comment (miles-grant-ibigroup) [inline lib/components/narrative/narrative-itineraries.js]:
```suggestion
        const transitLegs = cur.legs.filter((leg) => isTransitLeg(leg))
```

File: lib/util/itinerary.tsx
Comment (miles-grant-ibigroup) [inline lib/util/itinerary.tsx]: Please do the same shorthand as above

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
I trusted you! Post lunch I'm seeing this with new clairty

File: lib/components/narrative/default/itinerary-description.tsx
Comment (daniel-heppner-ibigroup) [reply lib/components/narrative/default/itinerary-description.tsx:35]: time to get over your fears

File: lib/components/narrative/default/itinerary-summary.js
Comment (daniel-heppner-ibigroup) [reply lib/components/narrative/default/itinerary-summary.js:100]: maybe, but maybe not

**Comment by daniel-heppner-ibigroup** [comment]
all the problems are fixed

File: lib/components/narrative/default/itinerary-description.tsx
Comment (miles-grant-ibigroup) [reply lib/components/narrative/default/itinerary-description.tsx:35]: 🤬

File: lib/components/narrative/default/itinerary-summary.js
Comment (miles-grant-ibigroup) [reply lib/components/narrative/default/itinerary-summary.js:100]: 🤬

File: lib/actions/field-trip.js
Comment (miles-grant-ibigroup) [inline lib/actions/field-trip.js]: We can use shorthand

File: lib/actions/field-trip.js
Comment (miles-grant-ibigroup) [inline lib/actions/field-trip.js]: We can use the shorthand

File: lib/components/narrative/narrative-itineraries.js
Comment (miles-grant-ibigroup) [inline lib/components/narrative/narrative-itineraries.js]: We can use the shorthand!

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
It can be swaggier

**Review by miles-grant-ibigroup — APPROVED** [review]
Excited for the second reviewer

File: lib/components/narrative/default/itinerary-description.tsx
Comment (amy-corson-ibigroup) [inline lib/components/narrative/default/itinerary-description.tsx:68]: Maybe make this check a variable? It's used twice, here and 35

**Review by amy-corson-ibigroup — COMMENTED** [review]
Uh well it does work.... I will be honest I am missing the logic of why we need this in core utils? Why import a function when we can just use `leg.isTransit`

File: lib/components/narrative/default/itinerary-description.tsx
Comment (daniel-heppner-ibigroup) [reply lib/components/narrative/default/itinerary-description.tsx:68]: I thought about that too but decided against it because it's a simple check anyway

**Review by amy-corson-ibigroup — APPROVED** [review]
I guess

**Pull request merged**
